google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java

```diff
@@ -575,6 +583,145 @@ public Builder toBuilder() {
     return builder.build();
   }
 
+  /**
+   * Customized serializers to match the expected format for timestamp, source location and request
+   * method
+   */
+  static final class InstantSerializer implements JsonSerializer<Instant> {
+    @Override
+    public JsonElement serialize(
+        Instant src, java.lang.reflect.Type typeOfSrc, JsonSerializationContext context) {
+      return new JsonPrimitive(src.toString());
```

Comment:
This gson "contract" guarantees the expected behavior. I think that extra checks are not necessary because it would mean incorrect operation of the gson which is a bigger problem.